ports/esp8266/modesp.c

```diff
@@ -203,6 +203,15 @@ STATIC mp_obj_t esp_neopixel_write_(mp_obj_t pin, mp_obj_t buf, mp_obj_t is800k)
 }
 STATIC MP_DEFINE_CONST_FUN_OBJ_3(esp_neopixel_write_obj, esp_neopixel_write_);
 
+STATIC mp_obj_t esp_neopixel_write_frac_(mp_obj_t pin, mp_obj_t buf, mp_obj_t is800k, mp_obj_t frac) {
+    mp_buffer_info_t bufinfo;
+    mp_get_buffer_raise(buf, &bufinfo, MP_BUFFER_READ);
+    esp_neopixel_write_frac(mp_obj_get_pin_obj(pin)->phys_port,
+			    (uint8_t*)bufinfo.buf, bufinfo.len, mp_obj_is_true(is800k), mp_obj_get_float(frac));
+    return mp_const_none;
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_4(esp_neopixel_write_frac_obj, esp_neopixel_write_frac_);
```

Comment:
Instead of this the standard way to do it (for any number of args above 3) is `STATIC MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN(esp_neopixel_write_frac_obj, 4, 4, esp_neopixel_write_frac_);` and the signature of the function is `(size_t nargs, const mp_obj_t *args)`.

Reply:
Thanks!